snapcraft/internal/lxd/_containerbuild.py

```diff
@@ -75,6 +75,8 @@ def __init__(self, *, output, source, project_options,
         self._lxd_common_dir = os.path.expanduser(
             os.path.join('~', 'snap', 'lxd', 'common'))
         os.makedirs(self._lxd_common_dir, exist_ok=True)
+        # Work-around for https://github.com/lxc/lxd/issues/4183
+        os.environ['SNAP'] = ''
```

Comment:
This feels a little heavy-handed, would it not be better to do this temporarily just whenever we shell out to lxc?